File: NeetCodeIO/704BinarySearch/BinarySearch.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        // Binary Search -> Time:O(log(n)), Space:O(1)
        int mid, start = 0, end = nums.size()-1;
        while (start <= end) {
            mid = (start + end) / 2;
            if (target > nums[mid]) {
                start = mid + 1;
            }
            else if (target < nums[mid]) {
                end = mid - 1;
            }
            else {
                return mid;
            }
        }
        return -1;
    }
};

File: NeetCodeIO/704BinarySearch/BinarySearch.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        # Binary Search -> Time:O(log(n)), Space:O(1)
        start, end = 0, len(nums)-1
        while start <= end:
            mid = end+start // 2
            if nums[mid] < target:
                start = mid + 1
            elif nums[mid] > target:
                end = mid - 1
            else:
                return mid
        return -1
